chain/indexer/src/streamer/streamer.rs

```diff
@@ -215,16 +241,18 @@ pub(crate) async fn start(
         };
 
         let latest_block_height = block.header.height;
-        if last_synced_block_height == 0 {
-            last_synced_block_height = latest_block_height;
+        if last_synced_block_height.is_none() {
+            db.put(b"last_synced_block_height", &latest_block_height.to_string()).unwrap();
+            last_synced_block_height = Some(latest_block_height);
         }
```

Comment:
It is better to avoid unnecessary `.unwrap()`s as those are potential places for crashes. Embrace the following pattern:

```suggestion
        let start_syncing_block_height = if let Some(last_synced_block_height) = last_synced_block_height {
            last_synced_block_height + 1
        } else {
            match indexer_config.sync_mode {
                crate::SyncModeEnum::FromInterruption => match db.get(b"last_synced_block_height").unwrap() {
                    Some(value) => String::from_utf8(value).unwrap().parse::<u64>().unwrap(),
                    None => latest_block_height,
                },
                crate::SyncModeEnum::LatestSynced => latest_block_height,
                crate::SyncModeEnum::BlockHeight(height) => Some(height),
            }    
        };
```

This also resolves the issue of Height(0) & Height(1) mentioned above.

In general, try to have only a single place where you update the value (`last_synced_block_height` and its DB value should be updated in a single place to make things easier to reason about).